.CMake/alg_support.cmake

```diff
@@ -15,7 +15,10 @@ if(NOT WIN32)
 else()
     option(OQS_USE_OPENSSL "Enable OpenSSL usage" OFF)
 endif()
-cmake_dependent_option(OQS_USE_AES_OPENSSL "" ON "OQS_USE_OPENSSL" OFF)
+
+# Use OpenSSL's AES only if no AESNI and x86 dist build is used.
+# Reason: AESNI implementation better fits our incremental API.
+cmake_dependent_option(OQS_USE_AES_OPENSSL "" ON "OQS_USE_OPENSSL; NOT OQS_DIST_X86_64_BUILD; NOT OQS_USE_AES_INSTRUCTIONS" OFF)
```

Comment:
I'm at a loss understanding this code: What defines exactly need to be set to disable AESNI? See #1229 to document this (as and when merged).

Reply:
> See #1229 to document this (as and when merged).

thx, will add documentation once #1229 is merged.

Reply:
To test different variants I was directly editing alg_support.cmake.

Looks like turning on/off `OQS_USE_${CPU_EXTENSION}_INSTRUCTIONS` via command line options will be overridden by the code here:
https://github.com/open-quantum-safe/liboqs/blob/e37ad5b5d8a0c7cc19a8a148c7adce710a010b6b/.CMake/gcc_clang_intrinsics.cmake#L14-L16
It should probably guarded like this:

```
foreach(CPU_EXTENSION ${RUN_OUTPUT})
    if (NOT DEFINED OQS_USE_${CPU_EXTENSION}_INSTRUCTIONS)
        set(OQS_USE_${CPU_EXTENSION}_INSTRUCTIONS ON)
     endif()
endforeach()
```

Reply:
> added a commit with this change

Thanks!